spring-integration-jms/src/main/java/org/springframework/integration/jms/JmsOutboundGateway.java

```diff
@@ -144,6 +145,12 @@
 
 	private volatile boolean requiresReply;
 
+	private long lastSend;
+
+	private volatile long idleReplyContainerTimeout;
```

Comment:
Strictly speaking, any field that can be changed by another thread must be `volatile` unless it is accessed synchronously.

I tend to relax this for framework setters (such `setApplicationContext`) because that won't be changed at runtime.

Fields that affect runtime behavior, such as this one, must be `volatile` if we want other threads to see the change (if it is changed later). With modern JVMs (as I understand it) reading volatile fields that rarely change has very little additional overhead.

`lastSend` and `idleTask` are only accessed within synchronized blocks, which itself ensures cache consistency, so they do not need to be volatile.